lib/model/use_cases/router_provider.dart

```diff
@@ -0,0 +1,33 @@
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:habit_app/model/use_cases/auth_providers.dart';
+import 'package:habit_app/presentation/screens/login_screen.dart';
+import 'package:habit_app/presentation/screens/main_screen.dart';
+
+final routerProvider = Provider.autoDispose<GoRouter>((ref) {
+  final userState = ref.watch(userStateProvider);
+
+  return GoRouter(
+    initialLocation: '/',
+    redirect: (context, state) {
+      final userStateValue = userState.value ?? (null, false);
+      final (user, verified) = userStateValue;
+
+      // ログインしていない場合はログイン画面へ遷移
+      if (user == null) {
```

Comment:
vefified がfalseの分岐が漏れているように見えます。
(nullになって画面遷移しない？)

```suggestion
      if (user == null || !vefified) {
```